### PR TITLE
move code linting to a stricter pep8-esque auto-formatting tool, black

### PR DESCRIPTION
**Pull request by ryanpetrello**
`black` (https://pypi.org/project/black/#description) is a strict superset of PEP8

It's also a tool that auto-formats your code on the fly for you based on its ruleset.

With this PR, you can run `make check`, and any style issues will automatically be applied.

Additionally, with this PR, if you spin up the development environment using our make targets, you'll automatically get a pre-commit hook installed that automatically runs linting prior to commit.

If you don't like this behavior, or don't want it locally, you can:

```shell
~ export AWX_IGNORE_BLACK=1
```

...but it's important to note that we won't merge your PR if it doesn't adhere to our style guidelines (which will run automatically as part of pre-merge CI).

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -296,7 +284,7 @@ swagger: reports
 	fi; \
 	(set -o pipefail && py.test $(PYTEST_ARGS) awx/conf/tests/functional awx/main/tests/functional/api awx/main/tests/docs --release=$(VERSION_TARGET) | tee reports/$@.report)
 
-check: flake8 pep8 # pyflakes pylint
+check: black
```
Comment (jbradberry) [inline Makefile:291]: I'm opposed to making it mandatory.

File: awx/api/conf.py [hunk 2]
```diff
@@ -71,8 +77,7 @@
     required=False,
     default='',
     label=_('Login redirect override URL'),
-    help_text=_('URL to which unauthorized users will be redirected to log in. '
-                'If blank, users will be sent to the Tower login page.'),
+    help_text=_('URL to which unauthorized users will be redirected to log in. ' 'If blank, users will be sent to the Tower login page.'),
```
Comment (jbradberry) [inline awx/api/conf.py]: a bit of weirdness, it's not smart enough apparently to glue the substrings back together.

File: awx/api/fields.py [hunk 3]
```diff
@@ -16,10 +16,10 @@
 
 
 class NullFieldMixin(object):
-    '''
+    """
```
Comment (jbradberry) [inline awx/api/fields.py:19]: yesyesyes!  :100:

File: setup.cfg [hunk 4]
```diff
@@ -1,21 +0,0 @@
-[pep8]
```
Comment (jbradberry) [inline setup.cfg:1]: I'm warming to the idea of using this tool, but I think that a distinction between linting and formatting should be maintained.  We should probably not get rid of pep8 and flake8.  What will Zuul be doing, after all?

Comment (ryanpetrello) [reply setup.cfg:1]: We *could* make a different target that uses black for CI, e.g. running `black --check`

```shell
(set -o pipefail && black --skip-string-normalization --check --fast --line-length 160 awx awxkit awx_collection | tee reports/black.report)
would reformat /home/ryanpetrello/dev/awx/awx/main/tasks.py
Oh no! 💥 💔 💥
1 file would be reformatted, 809 files would be left unchanged.
make: *** [Makefile:275: black] Error 1
~/dev/awx echo $?
2
```

Comment (ryanpetrello) [reply setup.cfg:1]:
```
~ tox -e linters
GLOB sdist-make: /home/ryanpetrello/dev/awx/setup.py
linters inst-nodeps: /home/ryanpetrello/dev/awx/.tox/.tmp/package/1/awx-17.0.1.zip
WARNING: Discarding $PYTHONPATH from environment, to override specify PYTHONPATH in 'passenv' in your configuration.
linters installed: -f file:///Users/ryan/.pip/wheelhouse,appdirs==1.4.4,arrow==1.0.3,awx==17.0.1,black==20.8b1,click==7.1.2,Jinja2==2.11.3,jinja2-time==0.2.0,make==0.1.6.post2,MarkupSafe==1.1.1,mypy-extensions==0.4.3,pathspec==0.8.1,python-dateutil==2.8.1,PyYAML==5.4.1,regex==2021.3.17,six==1.15.0,toml==0.10.2,typed-ast==1.4.2,typing-extensions==3.7.4.3,yamllint==1.26.0
linters run-test-pre: PYTHONHASHSEED='1240567368'
linters run-test: commands[0] | make black --check
(set -o pipefail && black --skip-string-normalization --check --fast --line-length 160 awx awxkit awx_collection | tee reports/black.report)
would reformat /home/ryanpetrello/dev/awx/awx/main/tasks.py
Oh no! 💥 💔 💥
1 file would be reformatted, 809 files would be left unchanged.
make: *** [Makefile:275: black] Error 1
ERROR: InvocationError for command /usr/bin/make black --check (exited with code 2)```

File: Makefile (hunk 1 above)
Comment (ryanpetrello) [reply Makefile:291]: Can you clarify this a bit?  Making what mandatory?

**Review by rooftopcellist — APPROVED** [review]

**Review by fosterseth — APPROVED** [review]

**Review by shanemcd — APPROVED** [review]

**Review by rebeccahhh — APPROVED** [review]

**Pull request merged**